bot/exts/moderation/stream.py

```diff
@@ -0,0 +1,149 @@
+        self.reload_task.cancel()
+        self.reload_task.add_done_callback(lambda _: self.scheduler.cancel_all())
+
+    async def _remove_streaming_permission(self, member: discord.Member) -> None:
+        """Remove the streaming permission from the given Member."""
+        await self.task_cache.delete(member.id)
+        await member.remove_roles(discord.Object(Roles.video), reason="Streaming access revoked")
+
+    async def _reload_tasks_from_redis(self) -> None:
+        """Reload outstanding tasks from redis on startup, delete the task if the member has since left the server."""
+        await self.bot.wait_until_guild_available()
+        items = await self.task_cache.items()
+        for key, value in items:
+            member = self.bot.get_guild(Guild.id).get_member(key)
+
+            if not member:
+                try:
+                    member = await self.bot.get_guild(Guild.id).fetch_member(key)
+                except discord.errors.NotFound:
+                    log.debug(
+                        f"Member {key} left the guild before we could scheudle "
+                        "the revoking of their streaming permissions."
+                    )
+                    await self.task_cache.delete(key)
+                    continue
+                except discord.HTTPException as e:
+                    log.exception(f"Exception while trying to retrieve member {key} from discord\n{e}")
+                    continue
+            revoke_time = datetime.datetime.utcfromtimestamp(value)
+            log.debug(f"Scheduling {member} ({member.id}) to have streaming permission revoked at {revoke_time}")
+            self.scheduler.schedule_at(
+                revoke_time,
+                key,
+                self._remove_streaming_permission(member)
+            )
+
+    @commands.command(aliases=("streaming",))
+    @commands.has_any_role(*STAFF_ROLES)
+    async def stream(self, ctx: commands.Context, user: discord.Member, duration: Expiry = None) -> None:
+        """
+        Temporarily grant streaming permissions to a user for a given duration.
+
+        A unit of time should be appended to the duration.
+        Units (∗case-sensitive):
+        \u2003`y` - years
+        \u2003`m` - months∗
+        \u2003`w` - weeks
+        \u2003`d` - days
+        \u2003`h` - hours
+        \u2003`M` - minutes∗
+        \u2003`s` - seconds
+
+        Alternatively, an ISO 8601 timestamp can be provided for the duration.
+        """
+        log.trace(f"Attempting to give temporary streaming permission to {user} ({user.id}).")
+        # if duration is none then calculate default duration
+        if duration is None:
+            now = datetime.datetime.utcnow()
+            duration = now + datetime.timedelta(minutes=VideoPermission.default_permission_duration)
+
+        # Check if user already has streaming permission
+        already_allowed = any(Roles.video == role.id for role in user.roles)
```

Comment:
```suggestion
        already_allowed = user._roles.has(Roles.video)
```
This should be changed in other places too.

Reply:
Due to the private nature of this API, it carries risks to use.

We were indecisive in whether to use it or not, so I'm going to leave it as-is or now.